### ISSUE TITLE
Put training loop into NF class

### ISSUE THREAD

**Issue by kazewong** [kazewong/flowMC#158]
Currently the training loop is constructed outside the normalizing flow model.

Since the function `make_training_loop` only has one argument, which is the optix optimizer, it is should be reasonably easy to incorporate the `train_flow` function into the normalizing flow model, so in the ideal case, the user can do something like `nf_model.fit(data)`.
